### PR TITLE
HPCC-32164 Minor improvements to new filename gathering code

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith ignore whitespace when comparing

File: ecl/hqlcpp/hqlcpp.ipp [hunk 1]
```diff
@@ -2146,19 +2146,23 @@ protected:
...
-            if (isSigned)
-                flags |= SummaryFlags::IsSigned;
-            if (map.find(name) == map.end())
-                map[name] = flags;
-            else
-                map[name] = map[name] & flags;
-        }
+        if (type == SummaryType::None)
+            return;
+        //Spill files are meaningless in roxie, and no current benefit in recording them for hthor/thor
```
Comment (jakesmith) [inline ecl/hqlcpp/hqlcpp.ipp:2151]: are SummaryType::JobTemp's useful?

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good - 1 question.

File: ecl/hqlcpp/hqlcpp.ipp (hunk 1 above)
Comment (ghalliday) [reply ecl/hqlcpp/hqlcpp.ipp:2151]: They might be - in that they are written to persistent storage, so they are potentially noteworthy.  We can always remove recording them at a later date, and they will not take up a significant amount of space.

File: common/workunit/workunit.cpp [hunk 2]
```diff
@@ -8748,9 +8748,12 @@ bool CLocalWorkUnit::getSummary(SummaryType type, SummaryMap &map) const
     StringArray s;
     {
         CriticalBlock block(crit);
-        const char *list = p->queryProp(xpath);
-        if (!list)
+        IPropertyTree * match = p->queryPropTree(xpath);
+        if (!match)
             return false;
```
Comment (jakesmith) [inline common/workunit/workunit.cpp:8753]: does it make sense to return false if no xpath match ,but true (below) if !list ?
maybe it does - false means bad format (given other return false)

**Review by jakesmith — APPROVED** [review]
@ghalliday - approved - 1 side question

File: common/workunit/workunit.cpp (hunk 2 above)
Comment (ghalliday) [reply common/workunit/workunit.cpp:8753]: false means this workunit did not record the information.  true means it recorded it, but it was empty.
I will add a comment to clarify.

**Pull request merged**